### PR TITLE
fix: marketing link tracking

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-343] 

### Description

Fixes marketing link tracking - tested on iOS ✅
Fixes double tracking on normal deep links for logged out users - tested on iOS ✅

More info on decisions can be found [here](https://artsy.slack.com/archives/C03NF9TKR/p1697645671461019)
and for the bonus bug fix ar

> [!WARNING]
> Would like for at least 1 person to test with any `click.artsy.net` link from an email with an android device if possible since yarn open-url doesn't work with android 

|Before|After|
|---|---|
|logged in||
|<video src="https://github.com/artsy/eigen/assets/21178754/f800d006-3cc0-4b7e-9fe8-125a1ef3eca4" width="200" />  |<video src="https://github.com/artsy/eigen/assets/21178754/82113844-2044-4a14-b735-ec2029cc7ad9" width="200" /> |
|logged out|| 
|<video src="https://github.com/artsy/eigen/assets/21178754/8c799ded-0fcc-4f4b-9b31-f4eb93ae76f8" width="200" /> |<video src="https://github.com/artsy/eigen/assets/21178754/710fafb7-a854-4fb4-bbe3-ef6e067735e8" width="200" /> |

> [!NOTE]
> Also fixes a double tracking bug that we had for logged out users - removed tracking from inside `useEffect` (see more context on this [slack thread](https://artsy.slack.com/archives/CA8SANW3W/p1697734806237409)

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- marketing link tracking - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-343]: https://artsyproduct.atlassian.net/browse/PHIRE-343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/utils/useDeepLinks.ts [hunk 1]
```diff
@@ -30,10 +30,8 @@ export function useDeepLinks() {
   }, [isHydrated, isLoggedIn])
 
   const handleDeepLink = (url: string) => {
-    // These will be redirected, avoided double tracking
-    if (!url.includes("click.artsy.net")) {
-      trackEvent(tracks.deepLink(url))
-    }
+    // track deep link tap
+    trackEvent(tracks.deepLink(url))
```
Comment (gkartalis) [inline src/app/utils/useDeepLinks.ts:34]: right now if a user is logged out we track the deeplink tap even if they never actually get navigated to the sceen. Do we want to change that to track [right before navigating to the page](https://github.com/artsy/eigen/compare/main...gkartalis/deeplink-tracking#diff-830da6cccfa3b0dcbc39c21db3cafbe2e916ec3c647967220620292f6785cb5fR36-R40) ?

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c9c643865330d6f2076a84dcc0ef3d1969ba1db4
</p>

**Comment by gkartalis** [comment]
asked here for the deeplink payload https://artsy.slack.com/archives/C03NF9TKR/p1697645671461019

File: src/app/utils/useDeepLinks.ts [hunk 2]
```diff
@@ -30,10 +30,8 @@ export function useDeepLinks() {
   }, [isHydrated, isLoggedIn])
 
   const handleDeepLink = (url: string) => {
-    // These will be redirected, avoided double tracking
-    if (!url.includes("click.artsy.net")) {
```
Comment (MounirDhahri) [inline src/app/utils/useDeepLinks.ts:34]: **issue:** do we really want to track links that look like `click.artsy.net` I find that these links are not actionable and are hard to understand without deobfuscating their payload.
I think we can make this slightly better for our data team by using the same helper method we're using here in order to tell what's the deep link URL and track that instead
https://github.com/artsy/eigen/blob/f1f5a168a9e7e020e7ddb8b406c82a41b5908222/src/app/system/navigation/navigate.ts#L69-L75

Comment (gkartalis) [reply src/app/utils/useDeepLinks.ts:34]: Waiting for the analytics team to reply on this question opened a thread yesterday [here](https://artsy.slack.com/archives/C03NF9TKR/p1697645671461019)

Comment (gkartalis) [reply src/app/utils/useDeepLinks.ts:34]: I guess if we decide on tracking the deobfuscated link we can leave this hook as it was and add tracking [here](https://artsy.slack.com/archives/C03NF9TKR/p1697645671461019) after we get the url

File: src/app/utils/useDeepLinks.ts [hunk 3]
```diff
@@ -29,21 +30,37 @@ export function useDeepLinks() {
...
-  const handleDeepLink = (url: string) => {
-    // These will be redirected, avoided double tracking
-    if (!url.includes("click.artsy.net")) {
-      trackEvent(tracks.deepLink(url))
+  const handleDeepLink = async (url: string) => {
+    let targetURL
+
+    // If the url is a marketing url, we need to fetch the redirect
+    try {
+      targetURL = await fetch(url)
```
Comment (MounirDhahri) [inline src/app/utils/useDeepLinks.ts:38]: we probably only need to do this if  `!url.includes("click.artsy.net"))`

**Review by brainbicycle — APPROVED** [review]
nice 🥇

**Review by MounirDhahri — APPROVED** [review]
Niice

**Pull request merged**